support MulNoNan for TF

ahmadchalhoub opened the pull request:
### Details:
Implementing support for the MulNoNan op in TensorFlow

### Tickets:
 - Closes #20758

rkazants commented:
Hi @ahmadchalhoub, you did extra changes - I believe:) Please leave only ones for TF FE.

Best regards,
Roman

ahmadchalhoub commented:
Hi @rkazants, thanks for the feedback. I updated the tests, which I think fixes things hopefully.
Btw, since the [initial](https://github.com/openvinotoolkit/openvino/pull/21459/commits/d651d7721da44bcae5ebb7d155a6a702aeb7f22e) commit had changes to extra unnecessary files, I updated that initial commit to remove those file changes, and then pushed the updated test changes in the [second](https://github.com/openvinotoolkit/openvino/pull/21459/commits/9e2ce484c74604012b8800b4bff6a54a0c5ce4e5) commit.

I am not sure if this is how it should be done. Please let me know if you have any feedback :) 

Best,
Ahmad

rkazants commented:
build_jenkins

rkazants commented on src/frontends/tensorflow_common/src/op/mul_no_nan.cpp:
```diff
@@ -0,0 +1,40 @@
+using namespace ov::op;
+
+namespace ov {
+namespace frontend{
+namespace tensorflow{
+namespace op{
```
```suggestion
namespace frontend {
namespace tensorflow {
namespace op {
```

rkazants commented on src/frontends/tensorflow_common/src/op/mul_no_nan.cpp:
```diff
@@ -0,0 +1,40 @@
+    auto y = node.get_input(1);
+
+    // prepare zero constants of the same type as the inputs
+    auto zero = make_shared<v0::Constant>(element::f32, Shape{}, 0.0f)->output(0);
+    zero = make_shared<v1::ConvertLike>(zero, y);
```
please use routine from utils:

```
auto const_zero = create_same_type_const_scalar<int32_t>(x, 0);
```

Do not forget to include `utils.hpp`

ahmadchalhoub replied:
@rkazants thanks for the review! Just updated the requested change.

rkazants reviewed (approved):
Looks good. Left one comment

rkazants commented:
build_jenkins

rkazants commented on src/frontends/tensorflow_common/src/op/mul_no_nan.cpp:
```diff
@@ -0,0 +1,38 @@
+#include "common_op_table.hpp"
+#include "utils.hpp"
```
```suggestion
```

rkazants commented on src/frontends/tensorflow_common/src/op/mul_no_nan.cpp:
```diff
@@ -0,0 +1,37 @@
+#include "common_op_table.hpp"
+#include "openvino/op/multiply.hpp"
+#include "openvino/op/select.hpp"
+
```
```suggestion
#include "utils.hpp"
```

rkazants commented on src/frontends/tensorflow_common/src/op/mul_no_nan.cpp:
```diff
@@ -0,0 +1,38 @@
+#include "common_op_table.hpp"
```
```suggestion
// Copyright (C) 2018-2023 Intel Corporation
// SPDX-License-Identifier: Apache-2.0
//

#include "common_op_table.hpp"
```

rkazants commented on src/frontends/tensorflow_common/src/op/mul_no_nan.cpp:
```diff
@@ -0,0 +1,42 @@
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "common_op_table.hpp"
```
```suggestion
#include "common_op_table.hpp"
#include "openvino/op/equal.hpp"
```

rkazants commented:
build_jenkins
